src/util/psbt/map/output.rs

```diff
@@ -121,37 +121,19 @@ impl TapTree {
 
     /// Returns iterator for a taproot script tree, operating in DFS order over leaf depth and
     /// leaf script pairs.
-    pub fn iter(&self) -> TapTreeIter {
+    pub fn script_leaves(&self) -> core::slice::Iter<ScriptLeaf> {
         self.into_iter()
     }
 }
 
-/// Iterator for a taproot script tree, operating in DFS order over leaf depth and
-/// leaf script pairs.
-pub struct TapTreeIter<'tree> {
-    leaf_iter: core::slice::Iter<'tree, LeafInfo>,
-}
-
-impl<'tree> Iterator for TapTreeIter<'tree> {
-    type Item = (u8, &'tree Script);
-
-    fn next(&mut self) -> Option<Self::Item> {
-        self.leaf_iter.next().map(|leaf_info| {
-            (leaf_info.merkle_branch.as_inner().len() as u8, &leaf_info.script)
-        })
-    }
-}
-
 impl<'tree> IntoIterator for &'tree TapTree {
-    type Item = (u8, &'tree Script);
-    type IntoIter = TapTreeIter<'tree>;
+    type Item = &'tree ScriptLeaf;
+    type IntoIter = core::slice::Iter<'tree, ScriptLeaf>;
```

Comment:
Can I pls address this in the follow-up PR? This will not be API breaking, so cen go into a minor version. I just do not want to miss the deadline the taproot release by discarding @sanket1729 review

Reply:
Changing from `core::slice::Iter` to our type definitely *is* API breaking.